lib/db_connection/ownership/manager.ex

```diff
@@ -66,29 +66,29 @@ defmodule DBConnection.Ownership.Manager do
             mode: mode, mode_ref: nil, ets: ets, log: log}}
   end
 
-  def handle_call({:mode, {:shared, pid}}, _from, %{mode: {:shared, current}} = state) do
+  def handle_call({:mode, {:shared, shared}}, {caller, _}, %{mode: {:shared, current}} = state) do
     cond do
-      pid == current ->
+      shared == current ->
         {:reply, :ok, state}
       Process.alive?(current) ->
         {:reply, :already_shared, state}
       true ->
-        share_and_reply(state, pid)
+        share_and_reply(state, shared, caller)
     end
   end
-  def handle_call({:mode, {:shared, pid}}, _from, state) do
-    share_and_reply(state, pid)
+  def handle_call({:mode, {:shared, shared}}, {caller, _}, state) do
+    share_and_reply(state, shared, caller)
   end
   def handle_call({:mode, mode}, _from, %{mode: mode} = state) do
     {:reply, :ok, state}
   end
-  def handle_call({:mode, mode}, _from, state) do
-    state = proxy_checkin_all(state, nil)
+  def handle_call({:mode, mode}, {caller, _from}, state) do
```

Comment:
nit: `_from` is not a good var name here as its only the `tag`, I think `_` more suitable

Reply:
```suggestion
  def handle_call({:mode, mode}, {caller, _}, state) do
```